### ISSUE TITLE
template instantiation generates warning: no matching class member found for

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10639]
template instantiation generates `warning: no matching class member found for`.  doxygen-1.8.11 does not generate the warning but  doxygen-1.8.12 and later does, include 1.11.0 (trunk).

doxygen-1.8.11:

```
doxygen -g
doxygen Doxyfile
```

doxygen-1.8.12:

```
doxygen -g
doxygen Doxyfile
...
Searching for member function documentation...
C:/dev/doxygen-1.8.12.windows.x64.bin/test.h:23: warning: no matching class member found for
  template <>
  HandleClass & Container< ClassImp >::pop() const
...
```

trunk:

```
doxygen -g
doxygen -x Doxyfile
# Difference with default Doxyfile 1.11.0 (5fa510c0901e7b679471473f88bdc50ec44a6bee*)
doxygen Doxyfile
...
Searching for member function documentation...
/mnt/c/dev/test_doxygen/test.h:23: warning: no matching class member found for
  template
  HandleClass & Container< ClassImp >::pop() const
Possible candidates:
  'HandleClass & Container< T >::pop() const' at line 18 of file /mnt/c/dev/test_doxygen/test.h
...
```

testcase file:
[testcase.zip](https://github.com/doxygen/doxygen/files/14224596/testcase.zip)

**Comment by albert-github** [comment]
Problem looks like to  be with the instantiation of `Container` in `Customer` i.e.
```
class Customer
{
    Container<ClassImp> impl_;
};
```

I think this problem has the same base issue as the issue in #10640.

**Comment by albert-github** [comment]
Apparently this issue is not the same as for issue #10640 as after its fix ([88dd1ae](https://github.com/doxygen/doxygen/commit/88dd1ae1b699557d942eeaabeffad3148a34fd54)) this problem still remains.

**Comment by doxygen** [comment]
@scott-zhong Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by scott-zhong** [comment]
@doxygen Yes, the commit addresses the problem.  Thank you!!!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
